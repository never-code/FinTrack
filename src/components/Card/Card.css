.card-container {
  height: 150px;
  width: 350px;
  border: 3px solid var(--text-color);
  border-radius: 15px;
  justify-content: unset;
  padding: 20px;
  margin-bottom: 3%;
}
.card-container .title {
  font-size: 23px;
  letter-spacing: 1px;
  font-weight: 600;
  height: 30px;
}
.card-container .sub-title {
  font-size: 12px;
  letter-spacing: 1px;
}
.card-container .value {
  font-size: 40px;
  font-weight: 600;
  letter-spacing: 3px;
  text-overflow: ellipsis;
  overflow: hidden;
}
.emi {
  background-color: #66dcc1;
  color: white;
}
.interest {
  color: #0170f3;
  background-color: #fbd94f;
}
.total {
  background-color: #0170f3;
  color: white;
}
@media only screen and (max-width: 300px) {
  .card-container {
    width: 200px;
    height: 100px;
 }
  .card-container .title {
    font-size: 12px;
 }
  .card-container .value {
    font-size: 16px;
 }
}
