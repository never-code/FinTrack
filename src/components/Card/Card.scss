.card-container {
  height: 150px;
  width: 350px;
  border: 3px solid var(--text-color);
  border-radius: 15px;
  justify-content: unset;
  padding: 20px;
  // margin: auto;
  margin-bottom: 3%;
  .title {
    font-size: 23px;
    letter-spacing: 1px;
    font-weight: 600;
    height: 30px;
  }
  .sub-title {
    font-size: 12px;
    letter-spacing: 1px;
  }
  .value {
    font-size: 40px;
    font-weight: 600;
    letter-spacing: 3px;
    text-overflow: ellipsis;
    overflow: hidden;
  }
}
.emi {
  background-color: #66DCC1;
  color: white;
}
.interest {
  color: #0170F3;
  background-color: #FBD94F;
}
.total {
  background-color: #0170F3;
  color: white;
}
@media only screen and (max-width: 300px) {
  .card-container {
    width: 200px;
    height: 100px;
    .title {
      font-size: 12px;
    }
    .value {
      font-size: 16px;
    }
  }
}