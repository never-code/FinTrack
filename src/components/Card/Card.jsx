import React from 'react';
import './Card.scss';

const Card = ({title, subTitle, value, className }) => (
  <div className={`card-container ${className}`}>
    <div className="title">{title}</div>
    {subTitle && <div className="sub-title"><span>{subTitle}</span></div>}
    <div className="value">{value}</div>
  </div>
);
export default Card;