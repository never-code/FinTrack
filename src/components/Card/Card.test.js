import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe('Header ', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Card
        title="Card Title"
        subTitle="This is Subtitle"
        value={123123123123}
        className="test"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
