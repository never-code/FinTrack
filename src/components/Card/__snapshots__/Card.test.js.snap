// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  should render successfully 1`] = `
<div>
  <div
    class="card-container test"
  >
    <div
      class="title"
    >
      Card Title
    </div>
    <div
      class="sub-title"
    >
      <span>
        This is Subtitle
      </span>
    </div>
    <div
      class="value"
    >
      123123123123
    </div>
  </div>
</div>
`;
